toolkit/src/components/cut/list-item/index.hbs

```diff
@@ -0,0 +1,24 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+}}
+{{#if (or this.href this.route @onClick)}}
+  <Hds::Interactive
+    class="cut-list-item active hds-surface-mid"
+    @href={{this.href}}
+    @isRouteExternal={{@isRouteExternal}}
+    @route={{this.route}}
+    @isHrefExternal={{@isHrefExternal}}
+    @query={{@query}}
+    @replace={{@replace}}
+    {{on 'click' this.onClickAction}}
+    ...attributes
+  >
+    {{yield (hash Content=(component "cut/list-item/content"))}}
+    {{yield (hash Action=(component "cut/list-item/action"))}}
```

Comment:
since the `Action` is itself just yielding other components, maybe consider simplifying things in this way:
```suggestion
    <div {{on 'click' this.onClickStopPropagation}}>
      {{yield
        (hash ActionButton=(component "hds/button" size="small")
          ActionDropdown=(component "hds/dropdown")
          ActionGeneric=(component "cut/list-item/action-generic")
        )}}
    </div>
```
This would also remove a level of indirection in the "consumers" code, from this:
```
<Cut::ListItem as |L|>
  <L.Content>Service a</L.Content>
  <L.Action as |A|>
    <A.Button @text='Invalidate'></A.Button>
  </L.Action>
</Cut::ListItem>
```
to this:
```
<Cut::ListItem as |L|>
  <L.Content>Service a</L.Content>
  <L.ActionButton @text='Invalidate'></L.ActionButton>
</Cut::ListItem>
```

(Up to you if it's better `<L.ActionButton>` or simply `<L.Button>`)